crypto/dh/dh_check.c

```diff
@@ -139,21 +137,7 @@ int DH_check(const DH *dh, int *ret)
             *ret |= DH_CHECK_INVALID_Q_VALUE;
         if (dh->j && BN_cmp(dh->j, t1))
             *ret |= DH_CHECK_INVALID_J_VALUE;
-
-    } else if (BN_is_word(dh->g, DH_GENERATOR_2)) {
-        l = BN_mod_word(dh->p, 24);
-        if (l == (BN_ULONG)-1)
-            goto err;
-        if (l != 11)
-            *ret |= DH_NOT_SUITABLE_GENERATOR;
-    } else if (BN_is_word(dh->g, DH_GENERATOR_5)) {
-        l = BN_mod_word(dh->p, 10);
-        if (l == (BN_ULONG)-1)
-            goto err;
-        if ((l != 3) && (l != 7))
-            *ret |= DH_NOT_SUITABLE_GENERATOR;
-    } else
-        *ret |= DH_UNABLE_TO_CHECK_GENERATOR;
+    }
```

Comment:
Should we perhaps check that `g != p - 1`? Right now we're only checking for `g < p`.

Reply:
Yes, not 0, 1, p-1, not negative, not larger than p-1, etc...
That is why I added a call to DH_check_params above.

Reply:
Did I miss the check for `p-1` somewhere?  I did not see it where I expected to see it, perhaps did not look in the right place.

Reply:
Are you sure you're using `DH_check_params()` correctly?  It does not return true/false for valid parameters, rather it returns a mask of misfeatures via its second argument...

Reply:
Is the idea to accumulate more "misfeature" bits before returning?

Reply:
Yes, I hope so.
The idea is to accumulate all bits from DH_check_params, and or in all bits from DH_check.

Reply:
Yes, exactly, in case DH_check_params adds more checks in the future.